src/Elasticsearch.Net/Responses/Special/BytesResponse.cs

```diff
@@ -12,9 +13,16 @@ public bool TryGetServerError(out ServerError serverError)
 		{
 			serverError = null;
 			if (this.Body == null || this.Body.Length == 0) return false;
-			using(var stream = new MemoryStream(this.Body))
-				serverError = ServerError.Create(stream);
-			return true;
+			try
```

Comment:
Should the try catch happen inside `ServerError.Create()`? Can we use the response mimetype as a hint as well here? e.g in the nginx proxy case we already know what we got back from the server is not json.

Reply:
++ to using the MIME type.

> Should the try catch happen inside ServerError.Create()

I think the try/catch should be in the `Try*` method, with the potential for `ServerError.Create()` to throw an exception if a `ServerError` cannot be created from the `Stream`. To my mind, this aligns with BCL `Try*` method semantics.

Reply:
TryX pattern would be even better :+1:

Reply:
Its more about the case where the MIME type informs us the response is not `JSON` we can omit the `TryCreate()` all together.

Reply:
So I understand correctly, are you suggesting 

1. if the MIME type is "application/json", call `ServerError.Create();` inside `BytesResponse`/`StringResponse`
2. if the MIME type is anything else, use `ServerError.TryCreate()`
?

Reply:
1. Call trycreate
2. Omit all together? A bad proxy could break this but a bad proxy is broken anyways.

Reply:
Have now added that check. Would you mind taking another look?